lib/usd/ui/layerEditor/stageSelectorWidget.cpp

```diff
@@ -112,25 +112,40 @@ void StageSelectorWidget::selectedIndexChanged(int index)
 void StageSelectorWidget::sessionStageChanged()
 {
     if (!_internalChange) {
-        auto index = _dropDown->findData(QVariant::fromValue(_sessionState->stage()));
+        size_t index = -1;
+        for (size_t i = 0; i < _dropDown->count(); i++) {
+            auto const& data = _dropDown->itemData(i);
+            auto        entry = data.value<SessionState::StageEntry>();
+            if (entry == _sessionState->stageEntry()) {
+                index = i;
+            }
+        }
         if (index != -1) {
             QSignalBlocker blocker(_dropDown);
             _dropDown->setCurrentIndex(index);
         }
     }
 }
 
-void StageSelectorWidget::stageRenamed(std::string const& name, PXR_NS::UsdStageRefPtr const& stage)
+void StageSelectorWidget::stageRenamed(
+    std::string const&              oldName,
+    SessionState::StageEntry const& renamedEntry)
 {
-    auto index = _dropDown->findData(QVariant::fromValue(stage));
+    size_t index = -1;
+    for (size_t i = 0; i < _dropDown->count(); i++) {
+        auto const& data = _dropDown->itemData(i);
+        auto        entry = data.value<SessionState::StageEntry>();
+        if (entry._displayName == oldName && entry._stage == renamedEntry._stage) {
```

Comment:
I don't think you should compare the _displayName to the 'old name'.  See comment in stageRenamedSignal
Please only compare the stage. Maybe send down the proxyShapePath instead if you need it identify the StageEntry when more than one proxy use the stage stage (which is the bug you're fixing I guess). The reason is that the display name is just that, something I trimmed to fit in a combo

Reply:
Fair enough, I responded there.